src/main/java/fi/aalto/cs/apluscourses/intellij/model/task/IntelliJComponentPresenterBase.java

```diff
@@ -56,4 +69,11 @@ private void removeHighlightInternal() {
   }
 
   protected abstract GenericHighlighter getHighlighter();
+
+  protected abstract boolean tryToShow();
+
+  @Override
+  public boolean isVisible() {
+    return getHighlighter() != null && getHighlighter().getComponent().isVisible();
```

Comment:
Store the highlighter in a local variable. Here, the second call could actually return null.

Reply:
i do that in later prs, should i change it here too

Reply:
i changed it here too